File: Python/EstruturaDeRepeticao/exercicio_34.py
"""
Os números primos possuem várias aplicações dentro da Computação, por exemplo na Criptografia.
Um número primo é aquele que é divisível apenas por um e por ele mesmo.
Faça um programa que peça um número inteiro e determine se ele é ou não um número primo.
"""
def prime_number(numero):
    if numero < 2:
        return False

    for i in range(2, int(numero ** 0.5) + 1):
        if numero % i == 0:
            return False

    return True

numero = int(input("Digite um número inteiro: "))

if prime_number(numero):
    print(f"{numero} é um número primo.")
else:
    print(f"{numero} não é um número primo.")

File: Java/EstruturaDeRepeticao/Exercicio34.java
// Os números primos possuem várias aplicações dentro da Computação, por exemplo na Criptografia.
// Um número primo é aquele que é divisível apenas por um e por ele mesmo.
// Faça um programa que peça um número inteiro e determine se ele é ou não um número primo.
package EstruturaDeRepeticao;
import java.util.Scanner;

public class Exercicio34 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite um número inteiro: ");
        int numero = scanner.nextInt();

        if (primeNumber(numero)) {
            System.out.println(numero + " é um número primo.");
        } else {
            System.out.println(numero + " não é um número primo.");
        }
        scanner.close();
    }

    public static boolean primeNumber(int numero) {
        if (numero <= 1) {
            return false;
        }

        for (int i = 2; i <= Math.sqrt(numero); i++) {
            if (numero % i == 0) {
                return false;
            }
        }

        return true;
    }
}
